src/Kopernicus/Configuration/Loader.cs

```diff
@@ -200,66 +200,91 @@ void IParserEventSubscriber.Apply(ConfigNode node)
             // Generates the system prefab from the configuration 
             void IParserEventSubscriber.PostApply(ConfigNode node)
             {
+                //Set of items seen while loading, to prevent duplicate loggers being created (throws IOExceptions)
+                HashSet<String> seen = new HashSet<String>();
+
                 // Dictionary of bodies generated
                 Dictionary<String, Body> bodies = new Dictionary<String, Body>();
 
                 // Load all of the bodies
                 foreach (ConfigNode bodyNode in node.GetNodes(bodyNodeName))
                 {
-                    // Create a logger for this body
-                    Logger bodyLogger = new Logger(bodyNode.GetValue("name") + ".Body");
-                    bodyLogger.SetAsActive();
+                    if ( seen.Contains(bodyNode.GetValue("name")) )
+                    {
+                        Logger.Default.Log("[Kopernicus::PostApply] Skipped duplicate body " + bodyNode.GetValue("name"));
+                        continue; //next body, please
+                    }
+                    else
+                    {
+                        seen.Add(bodyNode.GetValue("name"));
+                    }
 
-                    // Attempt to create the body
                     try
                     {
+                        // Create a logger for this body
+                        Logger bodyLogger = new Logger(bodyNode.GetValue("name") + ".Body");
+                        bodyLogger.SetAsActive();
+
+                        // Attempt to create the body
                         currentBody = new Body();
-                        Parser.LoadObjectFromConfigurationNode(currentBody, bodyNode, "Kopernicus");
+                        Parser.LoadObjectFromConfigurationNode(currentBody, bodyNode, "Kopernicus"); //logs to active logger
                         bodies.Add(currentBody.name, currentBody);
                         Events.OnLoaderLoadBody.Fire(currentBody, bodyNode);
                         Logger.Default.Log("[Kopernicus]: Configuration.Loader: Loaded Body: " + currentBody.name);
+
+                        // Restore default logger
+                        bodyLogger.Flush();
+                        Logger.Default.SetAsActive();
                     }
                     catch (Exception e)
                     {
-                        bodyLogger.LogException(e);
-                        Logger.Default.Log("[Kopernicus]: Configuration.Loader: Failed to load Body: " + bodyNode.GetValue("name"));
+                        //bodyLogger.LogException(e);  -- This would be nice but becomes a problem when duplicate loggers are created: can't log then
```

Comment:
Since you already check for a duplicated body before the try / catch block starts, this could stay as it is right now. The logger shouldn't throw because of duplicated names anymore, so it doesn't need to be created in a try / catch.